packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -355,6 +354,8 @@ Future<XcodeBuildResult> buildXcodeProject({
     'xcrun',
     'xcodebuild',
     '-configuration', configuration,
+    if (buildInfo.treeShakeIcons)
```

Comment:
Add this with the reset of the environment variables in buildCommands

Reply:
Did you have somewhere specific in mind? They seem sprinked throughout the rest of this function.

Reply:
in `buildCommands`, those get written to an xcconfig so builds through the xcode ui reference the variables correctly. You're adding it to the invocation which only takes effect when running through the flutter tool

Reply:
Moved this to the xcodeproject file where it writes to the generated xcconfig